internal/pkg/service/stream/api/service/sink.go

```diff
@@ -24,6 +24,12 @@ import (
 
 //nolint:dupl // CreateSource method is similar
 func (s *service) CreateSink(ctx context.Context, d dependencies.SourceRequestScope, payload *api.CreateSinkPayload) (*api.Task, error) {
+	// If user is not admin deny access for write
+	token := d.StorageAPIToken()
+	if token.Admin == nil {
+		return nil, svcerrors.NewForbiddenError(errors.New("only admin token can do write operations on streams"))
```

Comment:
I'd consider adding this error as a constant to be easily reusable...

And yeah as Petr said this should be covered by tests.

Reply:
Yes I will add tests, I wanted just to check that whether should I spend time to create some middleware for that but I am just newbie in goa and we would need to have dependend middleware on previous one. 

In case we do not want middleware I will add tests and error as constant

Reply:
I see some tests added but not the error constant?

Reply:
Ah I did not noticed that you want also contant from error. Adding it

Reply:
https://github.com/keboola/keboola-as-code/pull/2098/commits/f37419d1522b2621160ac31d7f7bc800df0285cb